libpromises/crypto.c

```diff
@@ -502,10 +502,29 @@ size_t CipherTextLengthMax(const EVP_CIPHER* cipher, size_t plaintext_length)
     return plaintext_length + (CipherBlockSizeBytes(cipher) * 2) - 1;
 }
 
+size_t PlainTextLengthMax(const EVP_CIPHER* cipher, size_t ciphertext_length)
+{
+    if(ciphertext_length <= 0)
+    {
+        ProgrammingError("PlainTextLengthMax: ciphertext_length is not positive (%ld)",
+                         ciphertext_length);
+    }
+
+    // check for potential integer overflow, leave some buffer
+    if(ciphertext_length > INT_MAX - 1024)
+    {
+        ProgrammingError("PlainTextLengthMax: ciphertext_length is too large (%ld)",
```

Comment:
It is today, but it is not guaranteed for tomorrow. `ProgrammingError()` is ideally unreachable code. It is not allowed to change program behaviour. That's why I'm asking whether these fixes are for actual crashes discovered during fuzzing.